src/states/node.rs

```diff
@@ -2283,7 +2277,59 @@ impl Node {
             }
             outbox.send_event(Event::Terminate);
         }
-        false
+    }
+
+    // Drop peers to which we think we have a direct or tunnel connection, but where Crust reports
+    // that we're not connected to the peer or tunnel node respectively.
+    fn purge_invalid_rt_entries(&mut self, outbox: &mut EventBox) -> Transition {
```

Comment:
Most of this function could be implemented in the peer manager, if you replaced `get_routing_peer_details` with `fn find_broken_peer_details(&self, crust_service: &Service) -> Vec<PeerId>;`